### PR TITLE
Add script to generate deployment proposal

### PR DESCRIPTION
**Pull request by fedgiac**
Adds a script that uses the function introduced in #46 in a more user-friendly way. This is the function that will be used to generate the DAO proposal.

Note: the settings file input is not validated, adding validation is left to a follow-up PR. If needed, we can validate the input by replacing the example file, which is tested for well-formattedness.

### Test Plan

Generate some claims with the claim generation script. Then try out the script:
```
$ npx hardhat deployment --network mainnet --claims ./output/test-claims/claims.csv --settings ./example/settings.json
Processing input files...
Generating Merkle proofs...
Clearing old files...
Saving generated data to file...
```

`mappings.json` and `chunks/` are the same as in the test deployment script. The other two files are:
[steps.json](https://github.com/gnosis/cow-token/files/7921362/steps.json.txt)
[addresses.json](https://github.com/gnosis/cow-token/files/7921363/addresses.json.txt).

These functions can be executed by an EOA in the given order to then check that everything is deployed as expected.

### CODE REVIEW

File: settings.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (josojo) [inline settings.json:1]: Is this file intended for commit? I would think the example is sufficient

File: src/tasks/deployment.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  });
+  await fs.rm(`${OUTPUT_FOLDER}/steps.json`, { recursive: true, force: true });
+  await fs.rm(`${OUTPUT_FOLDER}/claims.json`, { recursive: true, force: true });
+  await removeSplitClaimFiles(OUTPUT_FOLDER);
+
+  console.log("Saving generated data to file...");
+  await fs.mkdir(OUTPUT_FOLDER, { recursive: true });
+  await fs.writeFile(
+    `${OUTPUT_FOLDER}/addresses.json`,
+    JSON.stringify(addresses, undefined, 2),
```
Comment (josojo) [inline src/tasks/deployment.ts:105]: What is the `undefined` and `2` for?

File: src/tasks/deployment.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  );
+
+  console.log("Clearing old files...");
+  await fs.rm(`${OUTPUT_FOLDER}/addresses.json`, {
+    recursive: true,
+    force: true,
+  });
+  await fs.rm(`${OUTPUT_FOLDER}/steps.json`, { recursive: true, force: true });
+  await fs.rm(`${OUTPUT_FOLDER}/claims.json`, { recursive: true, force: true });
+  await removeSplitClaimFiles(OUTPUT_FOLDER);
```
Comment (josojo) [inline src/tasks/deployment.ts:99]: nit: tiny bit code duplication with test-deployment. Maybe we put these functions into a separate function. I need it my scripts as well..

**Review by josojo — COMMENTED** [review]
Really good.

I have one question below, once this is cleared up, i will approve

File: settings.json (hunk 1 above)
Comment (fedgiac) [reply settings.json]: No, good that you spotted it!

File: src/tasks/deployment.ts (hunk 2 above)
Comment (fedgiac) [reply src/tasks/deployment.ts:105]: It makes the JSON human readable by adding two spaces of indentations when needed. So, instead of `{"outer":{"inner":"text}}` we would get:
```
{
  "outer": {
    "inner":"text
  }
}
```

`undefined` is an unused parameter of that function.

File: src/tasks/deployment.ts (hunk 3 above)
Comment (fedgiac) [reply src/tasks/deployment.ts:99]: The only duplicated lines are:
```
  await fs.rm(`${OUTPUT_FOLDER}/claims.json`, { recursive: true, force: true });
  await removeSplitClaimFiles(OUTPUT_FOLDER);
```

The other two lines are new and there are different lines in the other script (because of `params.json`).
So I don't see a way to generalize it without making a generic function that takes files to be deleted.

Comment (fedgiac) [reply src/tasks/deployment.ts:99]: What is the other script for by the way? I would expect to reuse this when we introduce the xDAI deployment.

**Review by josojo — APPROVED** [review]
